packages/xo-web/src/xo-app/proxies/deploy-proxy.js

```diff
@@ -59,10 +65,13 @@ const Modal = decorate([
       idGatewayInput: generateId,
       idIpInput: generateId,
       idNetmaskInput: generateId,
+      idSelectNetwork: generateId,
       idSelectNetworkMode: generateId,
       idSelectSr: generateId,
 
       isStaticMode: (state, { value }) => value.networkMode === 'static',
+      networkPredicate: (state, { value }) =>
+        value.sr && (network => value.sr.$pool === network.$pool),
```

Comment:
Why is it necessary to check `value.sr`?

Reply:
`value.sr` can be undefined.

Reply:
Is the predicate used in that case?

Reply:
The predicate will be invoked even if the component is disabled.

Reply:
Ok.